app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/WifiDirectManager.java

```diff
@@ -0,0 +1,229 @@
+package org.kiwix.kiwixmobile.zim_manager.local_file_transfer;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.IntentFilter;
+import android.net.wifi.WpsInfo;
+import android.net.wifi.p2p.WifiP2pConfig;
+import android.net.wifi.p2p.WifiP2pDevice;
+import android.net.wifi.p2p.WifiP2pDeviceList;
+import android.net.wifi.p2p.WifiP2pInfo;
+import android.net.wifi.p2p.WifiP2pManager;
+import android.os.Build;
+import android.util.Log;
+import android.widget.Toast;
+import androidx.annotation.NonNull;
+import java.net.InetAddress;
+import org.kiwix.kiwixmobile.R;
+
+import static android.os.Looper.getMainLooper;
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.LocalFileTransferActivity.showToast;
+
+
+/**
+ * Manager for the Wifi-P2p API, used in the local file transfer module
+ * */
+public class WifiDirectManager implements WifiP2pManager.ChannelListener, WifiP2pManager.PeerListListener, WifiP2pManager.ConnectionInfoListener {
+
+  private static final String TAG = "WifiDirectManager";
+
+  @NonNull LocalFileTransferActivity activity;
+
+  /* Variables related to the WiFi P2P API */
+  private boolean wifiP2pEnabled = false; // Whether WiFi has been enabled or not
+  private boolean retryChannel = false;   // Whether channel has retried connecting previously
+
+  private WifiP2pManager manager;         // Overall manager of Wifi p2p connections for the module
+  private WifiP2pManager.Channel channel;
+  // Connects the module to device's underlying Wifi p2p framework
+
+  private final IntentFilter intentFilter = new IntentFilter();
```

Comment:
this does not need to be a field, just a local variable

Reply:
[Done](https://github.com/kiwix/kiwix-android/commit/083d7768b8fdf58803fc138bf0ab6c35be32f432#diff-7e01f6f425daaca67175b74c839c6af2R76)

Reply:
the field still exists commented out, I may be commenting too quickly as your work is ongoing

Reply:
Yeah, I generally keep major changes commented in the current commit so that in case of a bug in the future I can compare the new code with the old (commented) code at the same time (so I can edit while viewing changes, unlike in `diff`).

But I delete the commented portions soon after committing (like [here](https://github.com/kiwix/kiwix-android/pull/1276/commits/2e41f51fc84b3d1024ff400b7835d24a73a296af#diff-e1d59ae6e15dcb502e125fb6e243d15dL170)), so that subsequent commits don't have them.

Reply:
I've resolved this but maybe look at sourcetree (my preferred git tool) or some other diff tool.

Reply:
I had disabled the 'Git Integration' plugin to fasten Studio previously (Windows with 4 GB RAM), but forgot to change the preferences when I upgraded my RAM and shifted to Ubuntu
Its 'compare' feature is nice, I'll use that moving forward instead of comments :+1: